helm/hpcc/templates/eclccserver.yaml

```diff
@@ -81,11 +82,11 @@ data:
 {{ include "hpcc.addImageAttrs" (dict "root" $ "me" .) | indent 12 }}
             resources:
               requests:
```

Comment:
Should the whole section be replaced?  E.g. to allow for user-defined resources etc.

Reply:
It's not straight-forward to support user-defined resources & copying over whole section:
1) Each of these values (requests/memory, requests/cpu, limit/memory & limit/cpu) are generated by merging the values from workunit debug value and the values from ChildContainerResources/slaveContainerResources.  It won't know what to do with any user-defined values.
2) The indentation will need be handled.

To support user-defined resources, is not trivial.  I suggest that we add support for user-defined resources in the next release. @ghalliday what do you think?

Reply:
It may be same as Gavin is saying, but I think the generation of the whole resources: section should be generated here, if needed, and in other similar jobspec places. 

Any per-job overrides can still be picked up from properties and merged in, including any that a user has decided to provide.
That probably means changing the current search and replace code, to load the yaml into a property tree and setting the overrides into the right place in the property tree, as opposed to the search and replace technique currently used.

Reply:
I have looked into replacing the current search & replace with a PropertyTree.  I have found more complications.  runK8sJob/LaunchK8sJob are passed a list of string pairs that are search and replaced in the yaml.  For example, a valid call is runK8sJob( ... , { { "graphName", graphName} });

"%graphName" is replaced by the actual graph name.  The problem is that we do not know the path to the searched-for string - do we iterate over every element of the IPropertyTree examining the node value for the tag? Is there a more efficient way? @jakesmith